src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -603,4 +709,101 @@ private double getRandomPitch()
     {
         return PITCH_DIVIDER / (worker.getRNG().nextDouble() * PITCH_MULTIPLIER + BASE_PITCH);
     }
+
+
+
+    /**
+     * Class to hold information about required item for the guard.
+     *
+     */
+    public class GuardItemsNeeded
+    {
+    	/**
+    	 * Quantity required on the required
+    	 */
+    	private final int quantity;
+
+    	/**
+    	 * Min level the citizen has to be to required the item
+    	 */
+    	private final int minLevelRequired;
+
+    	/**
+    	 *Max level the citizen can be to required the item
+    	 */
+    	private final int maxLevelRequired;
+    	/**
+    	 * Item that is being required
+    	 */
+    	private final Item itemNeeded;
+    	/**
+    	 * Item type that is required
+    	 */
+    	private final EntityEquipmentSlot type;
+
+    	/**
+    	 * @param type		item type for the required item
+    	 * @param item		item that is being required
+    	 * @param quantity	quantity required for the item
+    	 * @param min		min level required to demand item
+    	 * @param max		max level that the item will be required
+    	 */
+    	public GuardItemsNeeded(final EntityEquipmentSlot type, final Item item, final int quantity, final int min, final int max)
+    	{
+    		this.type = type;
+    		this.minLevelRequired = min;
+    		this.maxLevelRequired = max;
+    		this.itemNeeded = item;
+    		this.quantity = quantity;
+    	}
+
+    	/**
+    	 * @return  item that is required in stack format
+    	 */
+    	public ItemStack getItemStackNeeded()
+    	{
+    		return new ItemStack(itemNeeded, quantity);
+    	}
+
+    	/**
+    	 * @return item that is required
+    	 */
+    	public Item getItemNeeded()
+    	{
+    		return itemNeeded;
+    	}
+
+    	/**
+    	 * @return min level for this item to be required
+    	 */
+    	public int getMinLevelRequired()
+    	{
+    		return minLevelRequired;
```

Comment:
the formatting in this class is way off as well. Did you get the minecolonies formatter to work already?